src/promptflow/promptflow/_trace/_start_trace.py

```diff
@@ -25,24 +26,44 @@
 time_delay = 10
 
 
-def start_trace():
+def start_trace(*, session: typing.Optional[str] = None, **kwargs):
     """Start a tracing session.
 
     This will capture OpenAI and prompt flow related calls and persist traces;
     it will also provide a UI url for user to visualize traces details.
 
     Note that this function is still under preview, and may change at any time.
     """
+    from promptflow._sdk._constants import ExperimentContextKey
     from promptflow._sdk._service.utils.utils import get_port_from_config
 
     pfs_port = get_port_from_config(create_if_not_exists=True)
     _start_pfs_in_background(pfs_port)
     _logger.debug("PFS is serving on port %s", pfs_port)
+
     # provision a session
-    session_id = _provision_session()
+    session_id = _provision_session(session_id=session)
     _logger.debug("current session id is %s", session_id)
-    # init the global tracer with endpoint, context (session, run, exp)
+
+    operation_context = OperationContext.get_instance()
+
+    # honor and set attributes if user has specified
+    attributes: dict = kwargs.get("attributes", None)
+    if attributes is not None:
+        for attr_key, attr_value in attributes.items():
+            operation_context._add_otel_attributes(attr_key, attr_value)
+
+    # prompt flow related, retrieve `experiment` and `referenced.line_run_id`
```

Comment:
if user calls `start_trace` at the very beginning, before the experiment, this will not take effect?

I think we still need somewhere to get from environ and set in operation context

cc. @wangchao1230

Reply:
what do you mean by before the experiment?